uiowa/uiowa
# Events pager

Resolves https://github.com/uiowa/uiowa/issues/5189

Announced a change to the API:
https://iowaweb.slack.com/archives/C0KMREKNU/p1658435749631079

**For this to merge, a deployment of `content_hub_api_total_results` on uiowa702 needs to happen first.**

# To Test

- Sync down a v2 site, no config errors.
- Sync down a v3 site, no config errors.
- `drush cset sitenow_events.settings base_endpoint https://content.dev.drupal.uiowa.edu/api/v1/` on each site you are testing.
- See that you are able to interact (no errors) with events block the same and that you get the correct number of results based on a basic filter (audience: everyone, number of items: 3)
- Edit existing events block, create new events block, see the show pager checkbox. interact with it as it relates to the number of results field.
- If pager is on, the number of results is the number of items per page. Test the pager to make sure the items change and match the same order that would occur if you turned the pager off and increased the number of items.
- Past events option should also work.
- Anonymous users should get cached queries after initial paging. Cache times are the same as before.

Interesting tidbit: I think we have events from 2009 to 2025 in the system now
